Added tests to close code coverage in CIM area

JamesWTruher opened the pull request:
Using our code coverage report to close some test gaps

Added CimAdapter.Tests for adaptation layer (code cribbed from BVT tests)
Added SessionOption.Tests.ps1 (Code cribbed from BVT tests)
Modified CimClass tests by adding more tests
Added CimSession.Tests for basic tests of New-CimSession cmdlet
Added CimInstance.Tests for basic tests of Get-CimInstance

JamesWTruher committed:
Added tests to close code coverage in CIM area

Added CimAdapter.Tests for adaptation layer (code cribbed from BVT tests)
Added SessionOption.Tests.ps1 (Code cribbed from BVT tests)
Modified CimClass tests by adding more tests
Added CimSession.Tests for basic tests of New-CimSession cmdlet
Added CimInstance.Tests for basic tests of Get-CimInstance

msftclas commented:
Hi __@JamesWTruher__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Jim Truher). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla.microsoft.com.
    </p>

TTYL, MSBOT;

JamesWTruher committed:
Add CI tags to tests

lzybkr reviewed (commented):
This PR uses multiple ways of skipping tests:
- Try/Finally
- BeforeAll/AfterAll
- Cloning/restoring PSDefaultParameterValues
- Setting/clearing "it:pending"

I'm confused - is there a good reason for these differences?

JamesWTruher commented:
why are you confused? these are all valid techniques for skipping or marking tests as pending. Are you suggesting that we may only use one way for marking test skip or pend/

lzybkr commented:
We provide guidance in our documentation on how to skip tests. If we don't follow that guidance, there should be a good reason.

Also - consistency makes reviewing changes easier - when we deviate from a common pattern, it requires more effort to validate that the code is correct.

JamesWTruher commented:
both try/finally and beforeall are allowed forms for skipping tests

lzybkr commented:
Your guidelines suggest only one style: https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/WritingPesterTests.md#skipping-tests-in-bulk

Barring other good technical reasons, I'd prefer consistency here.

iSazonov commented on test/powershell/engine/Remoting/SessionOption.Tests.ps1:
```diff
@@ -0,0 +1,55 @@
+    Describe " WSMan SessionOption object" {
+        It "The SessionOption type exists" {
+            "Microsoft.WSMan.Management.SessionOption" -as "Type" | Should Not BeNullOrEmpty
+        }
```
This test should be removed because any test below will check it.

iSazonov commented on test/powershell/Modules/CimCmdlets/CimSession.Tests.ps1:
```diff
@@ -0,0 +1,45 @@
+        It "A cim session can be removed" {
+            $sessionName = [guid]::NewGuid()
+            $session = New-CimSession -ComputerName . -name $sessionName
+            $sessiosn += $session
```
Here typo $sessio**SN**

iSazonov replied:
#Closed

iSazonov commented:
@JamesWTruher You are a virtuoso but I agree with @lzybkr it is not clear without comments why the guide ignored.

JamesWTruher committed:
unify test execution to use try/catch for marking tests as skipped

moved test which retrieves cimclass via method to feature as it is not
a common operation, this test can take some time as well

JamesWTruher committed:
Fix typo in test

JamesWTruher commented:
this should have all outstanding comments addressed

iSazonov commented:
@JamesWTruher If you use try/catch for skipping all test in a file 
probably we should write this in the documentation as the primary way?

iSazonov commented:
LGFM

JamesWTruher commented:
Can this now be merged?
